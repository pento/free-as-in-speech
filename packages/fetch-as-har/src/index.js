const staticPagesParserWix = () => {};

module.exports = {
	staticPagesParserWix,
};
