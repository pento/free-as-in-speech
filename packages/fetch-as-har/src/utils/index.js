module.exports = {
	IdFactory: require( './idfactory' ),
};
