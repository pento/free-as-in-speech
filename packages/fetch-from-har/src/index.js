/**
 * find the request in the HAR.
 *
 * @param {Object} HAR
 * @param {string} requestUrl
 * @param {Object} options
 */
function findRequestInHar( HAR, requestUrl, options ) {
	const queryComparison =
		options && 'function' === typeof options.queryComparison
			? options.queryComparison
			: ( requestValue, harValue ) => requestValue === harValue;
	const req = new URL( requestUrl );
	// lets do a first pass on it.
	const entries = HAR.log.entries.filter( ( e ) => {
		const u = e.request.url;
		if ( u.host !== req.host ) {
			return false;
		}
		if ( u.pathname !== req.pathname ) {
			return false;
		}
		if (
			options &&
			options.method &&
			e.request.method !== ( options.method || 'GET' )
		) {
			return false;
		}
		return true;
	} );

	if ( entries.length > 1 ) {
		let results = entries;
		// first lets filter on query params
		const withSameQueryString = entries.filter( ( e ) => {
			return e.request.queryString.every( ( qs ) => {
				return queryComparison(
					req.searchParams.get( qs.name ),
					qs.value,
					qs.name,
					e.request.url
				);
			} );
		} );
		if ( withSameQueryString.length > 0 ) {
			results = withSameQueryString;
		}
		if ( results.length > 1 && req.body ) {
			// then try to filter on body
			const withTheSameBody = entries.filter( ( e ) => {
				const data = e.request.postData;
				if ( data ) {
					return req.body === data.text;
				}
				return false;
			} );
			if ( withTheSameBody.length > 0 ) {
				results = withTheSameBody;
			}
		}
		return results;
	}
	return entries;
}

function fetchFromHAR( HAR, mockOptions ) {
	if (
		typeof HAR.log === 'undefined' ||
		typeof HAR.log.entries === 'undefined'
	) {
		HAR.log = { entries: {} };
	}

	HAR.log.entries.forEach( ( e ) => {
		e.request.url = new URL( e.request.url );
	} );

	return function ( url, options ) {
		return new Promise( ( resolve ) => {
			const entries = findRequestInHar(
				HAR,
				url,
				Object.assign( options, mockOptions )
			);
			let entry = {
				request: {
					url,
				},
				response: {
					status: 404,
					statusText: 'Not Found',
					content: {
						text: '',
					},
				},
			};
			if ( ! entries.length ) {
				if ( options && 'function' === typeof options.fallback ) {
					entry = options.fallback( url, entry );
				}
			} else {
				entry = entries[ 0 ];
			}

			const request = entry.request;
			const response = entry.response;
			resolve( {
				ok: ( response.status / 100 || 0 ) === 2, // 200-299
				statusText: response.statusText,
				status: response.status,
				url: response.redirectURL || request.url,
				text: () => Promise.resolve( response.content.text ),
				json: () =>
					Promise.resolve( response.content.text ).then( JSON.parse ),
				blob: () =>
					Promise.resolve( new Blob( [ response.content.text ] ) ),
				clone: response,
				headers: {
					keys: () =>
						response.headers.map( ( header ) => header.name ),
					entries: () =>
						response.headers.map( ( header ) => [
							header.name,
							header.value,
						] ),
					get: ( n ) => response.headers[ n.toLowerCase() ],
					has: ( n ) => n.toLowerCase() in response.headers,
				},
			} );
		} );
	};
}

module.exports = fetchFromHAR;
