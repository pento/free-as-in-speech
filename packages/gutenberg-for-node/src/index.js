const { JSDOM, ResourceLoader } = require( 'jsdom' );
const fetchNode = require( 'node-fetch' ).default;
const noop = function () {};

class Window {
	constructor( jsdomConfig = {} ) {
		const { proxy, strictSSL, userAgent } = jsdomConfig;
		const resources = new ResourceLoader( {
			proxy,
			strictSSL,
			userAgent,
		} );
		return new JSDOM(
			'',
			Object.assign( jsdomConfig, {
				resources,
			} )
		).window;
	}
}

global.window = new Window( { url: 'http://localhost' } );
global.document = window.document;
global.requestAnimationFrame = global.cancelAnimationFrame = noop;
global.navigator = window.navigator;
global.Mousetrap = {
	init: noop,
	prototype: { stopCallback: noop },
};
window.matchMedia = global.matchMedia = () => ( { addListener: noop } );
global.Node = window.Node;
window.fetch = window.fetch || fetchNode;
global.CSS = {
	supports: noop,
	escape: noop,
};
