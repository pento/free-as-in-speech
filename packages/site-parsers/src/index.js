const utils = require( './utils' );
const { staticPagesParser } = require( './parsers/wix' );

module.exports = {
	utils,
	staticPagesParserWix: staticPagesParser,
};
