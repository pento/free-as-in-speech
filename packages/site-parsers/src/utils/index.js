module.exports = {
	IdFactory: require( './idfactory' ),
	...require( './register-blocks' ),
};
