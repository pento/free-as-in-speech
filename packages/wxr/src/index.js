/**
 * External dependencies
 */
import moment from 'moment';
import { create } from 'xmlbuilder2';

const DATE_RFC2822 = 'ddd, DD MMM YYYY HH:mm:ss [GMT]';
const DATE_MYSQL = 'YYYY-MM-DD HH:mm:ss';

/**
 * Main WXR class.
 */
export default class WXR {
	/**
	 * Constructor.
	 */
	constructor() {
		this.wxr = create();

		this.addHeader();

		this.channel = this.wxr
			.ele( 'rss', {
				version: '2.0',
				'xmlns:excerpt': 'http://wordpress.org/export/1.2/excerpt/',
				'xmlns:content': 'http://purl.org/rss/1.0/modules/content/',
				'xmlns:wfw': 'http://wellformedweb.org/CommentAPI/',
				'xmlns:dc': 'http://purl.org/dc/elements/1.1/',
				'xmlns:wp': 'http://wordpress.org/export/1.2/',
			} )
			.ele( 'channel' );

		this.channel
			.ele( 'pubDate' )
			.txt( moment.utc().format( DATE_RFC2822 ) );
		this.channel.ele( 'wp:wxr_version' ).txt( '1.2' );
	}

	/**
	 * Adds instruction comments to the WXR file.
	 */
	addHeader() {
		const instructions = [
			'This is a WordPress eXtended RSS file generated by the "Free as in Speech" browser extension, as an export of your site.',
			"It contains information about your site's posts, pages, comments, categories, and other content.",
			'You may use this file to transfer that content from one site to another.',
			'This file is not intended to serve as a complete backup of your site.',
			'',
			'To import this information into a WordPress site follow these steps:',
			'1. Log in to that site as an administrator.',
			'2. Go to Tools: Import in the WordPress admin panel.',
			'3. Install the "WordPress" importer from the list.',
			'4. Activate & Run Importer.',
			'5. Upload this file using the form provided on that page.',
			'6. You will first be asked to map the authors in this export file to users',
			'   on the site. For each author, you may choose to map to an',
			'   existing user on the site or to create a new user.',
			'7. WordPress will then import each of the posts, pages, comments, categories, etc.',
			'   contained in this file into your site.',
		];

		instructions.forEach( ( instruction ) =>
			this.wxr.com( ` ${ instruction } ` )
		);
	}

	/**
	 * Sets the site meta data.
	 *
	 * @param {Object} metaData The site meta data object.
	 */
	setSiteMeta( metaData ) {
		// Define the site meta properties that can be passed, and what XML elements they map to.
		const metaMap = [
			{
				key: 'title',
				elements: [ 'title' ],
			},
			{
				key: 'description',
				elements: [ 'description' ],
			},
			{
				key: 'language',
				elements: [ 'language' ],
			},
			{
				key: 'url',
				elements: [ 'link', 'wp:base_site_url', 'wp:base_blog_url' ],
			},
		];

		// Set the corresponding XML elements for any meta data passed.
		metaMap.forEach( ( metaDef ) => {
			if ( metaData.hasOwnProperty( metaDef.key ) ) {
				metaDef.elements.forEach( ( element ) =>
					this.channel.ele( element ).txt( metaData[ metaDef.key ] )
				);
			}
		} );
	}

	/**
	 * Add a category to the export.
	 *
	 * @param {Object} category The category object.
	 */
	addCategory( category ) {
		const dataFilters = [
			{
				key: 'id',
				element: 'wp:term_id',
				filter: ( data ) => parseInt( data ),
			},
			{
				key: 'slug',
				element: 'wp:category_nicename',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'parent',
				element: 'wp:category_parent',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'name',
				element: 'wp:cat_name',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'description',
				element: 'wp:category_description',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
		];

		const categoryEl = this.channel.ele( 'wp:category' );

		dataFilters.forEach( ( filter ) => {
			if ( category.hasOwnProperty( filter.key ) ) {
				const data = filter.filter( category[ filter.key ] );
				if ( filter.cdata ) {
					categoryEl
						.ele( filter.element, filter.attributes )
						.dat( data );
				} else {
					categoryEl
						.ele( filter.element, filter.attributes )
						.txt( data );
				}
			}
		} );
	}

	/**
	 * Add a tag to the export.
	 *
	 * @param {Object} tag The tag object.
	 */
	addTag( tag ) {
		const dataFilters = [
			{
				key: 'id',
				element: 'wp:term_id',
				filter: ( data ) => parseInt( data ),
			},
			{
				key: 'slug',
				element: 'wp:tag_slug',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'name',
				element: 'wp:tag_name',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'description',
				element: 'wp:tag_description',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
		];

		const categoryEl = this.channel.ele( 'wp:tag' );

		dataFilters.forEach( ( filter ) => {
			if ( tag.hasOwnProperty( filter.key ) ) {
				const data = filter.filter( tag[ filter.key ] );
				if ( filter.cdata ) {
					categoryEl
						.ele( filter.element, filter.attributes )
						.dat( data );
				} else {
					categoryEl
						.ele( filter.element, filter.attributes )
						.txt( data );
				}
			}
		} );
	}

	/**
	 * Add an author to the export.
	 *
	 * @param {Object} author The author object.
	 */
	addAuthor( author ) {
		const keys = [
			'id',
			'login',
			'email',
			'display_name',
			'first_name',
			'last_name',
		];

		const authorEl = this.channel.ele( 'wp:author' );

		keys.forEach( ( key ) => {
			if ( author.hasOwnProperty( key ) ) {
				authorEl.ele( `wp:author_${ key }` ).txt( author[ key ] );
			}
		} );
	}

	/**
	 * Add a post to the export.
	 *
	 * @param {Object} post The post object.
	 */
	addPost( post ) {
		const dataFilters = [
			{
				key: 'id',
				element: 'wp:post_id',
				filter: ( data ) => parseInt( data ),
			},
			{
				key: 'title',
				element: 'title',
				filter: ( data ) => data.toString(),
			},
			{
				key: 'link',
				element: 'link',
				filter: ( data ) => data.toString(),
			},
			{
				key: 'date',
				element: 'pubDate',
				filter: ( data ) =>
					moment( data ).isValid()
						? moment( data ).utc().format( DATE_RFC2822 )
						: '',
			},
			{
				key: 'author',
				element: 'dc:creator',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'guid',
				element: 'guid',
				attributes: { isPermaLink: false },
				filter: ( data ) => data.toString(),
			},
			{
				key: 'content',
				element: 'content:encoded',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'excerpt',
				element: 'excerpt:encoded',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'date',
				element: 'wp:post_date',
				cdata: true,
				filter: ( data ) =>
					moment( data ).isValid()
						? moment( data ).utc().format( DATE_MYSQL )
						: '',
			},
			{
				key: 'date',
				element: 'wp:post_date_gmt',
				cdata: true,
				filter: ( data ) =>
					moment( data ).isValid()
						? moment( data ).utc().format( DATE_MYSQL )
						: '',
			},
			{
				key: 'status',
				element: 'wp:status',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'type',
				element: 'wp:post_type',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
			{
				key: 'sticky',
				element: 'wp:is_sticky',
				filter: ( data ) => parseInt( data ),
			},
			{
				key: 'comment_status',
				element: 'wp:comment_status',
				cdata: true,
				filter: ( data ) => data.toString(),
			},
		];

		const postEl = this.channel.ele( 'item' );

		dataFilters.forEach( ( filter ) => {
			if ( post.hasOwnProperty( filter.key ) ) {
				const data = filter.filter( post[ filter.key ] );
				if ( filter.cdata ) {
					postEl.ele( filter.element, filter.attributes ).dat( data );
				} else {
					postEl.ele( filter.element, filter.attributes ).txt( data );
				}
			}
		} );

		// Handle terms.
		if ( Array.isArray( post.terms ) ) {
			post.terms.forEach( ( term ) =>
				postEl
					.ele( 'category', {
						domain: term.type,
						nicename: term.slug,
					} )
					.dat( term.name )
			);
		}
	}

	/**
	 * Serialises the export into WXR XML.
	 */
	export() {
		return this.wxr.end( {
			allowEmptyTags: true,
			indent: '\t',
			prettyPrint: true,
		} );
	}
}
