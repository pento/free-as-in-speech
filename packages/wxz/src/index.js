/**
 * Internal dependencies
 */
const { WXZDriver: WXZ1Driver } = require( './1/index.js' );

const SUPPORTED_VERSIONS = [ 1 ];

const getWXZDriver = async ( wxzVersion, reset = false ) => {
	if ( ! SUPPORTED_VERSIONS.includes( wxzVersion ) ) {
		return;
	}

	let driver;

	switch ( wxzVersion ) {
		case 1:
			driver = new WXZ1Driver();
			break;
	}

	await driver.connect( { reset } );

	return driver;
};

module.exports = {
	getWXZDriver,
	SUPPORTED_VERSIONS,
};
