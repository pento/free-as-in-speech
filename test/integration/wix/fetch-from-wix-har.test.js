/**
 * WordPress dependencies
 */
const FDBFactory = require( 'fake-indexeddb/lib/FDBFactory' );
const { registerBlocks } = require( 'site-parsers' ).utils;
/**
 * Internal dependencies
 */
const fs = require( 'fs' );
const path = require( 'path' );
const fetchFromHAR = require( 'fetch-from-har' );
const getWXRFromWixHAR = require( '../../../bin/lib/get-wxr-from-wix-har' );
const { startExport } = require( '../../../source/services/wix' );

beforeEach( () => {
	window.indexedDB = new FDBFactory();
	require( '../../../packages/gutenberg-for-node/src/index' );
	registerBlocks();
} );

test.each( [
	[
		'empty.har',
		{
			initialState: {
				embeddedServices: {},
			},
			extractAll: true,
		},
		false,
		false,
	],
	[
		'wix-basic.har',
		{
			initialState: {
				embeddedServices: [
					'14bcded7-0066-7c35-14d7-466cb3f09103',
					'media-manager',
					'22bef345-3c5b-4c18-b782-74d4085112ff',
				].map( ( appDefinitionId ) => {
					return { appDefinitionId };
				} ),
			},
		},
		false,
		false,
	],
	[
		'easyblog.har',
		{
			initialState: {
				siteMetaData: {
					metaSiteId: 'ae5cf16f-06cb-499f-9fcd-6b08ba634e04',
				},
				embeddedServices: [
					'13d7e48f-0739-6252-8018-457a75beae4b',
				].map( ( appDefinitionId ) => {
					return { appDefinitionId };
				} ),
			},
		},
		false,
		true,
	],
	[
		'rockfield.har',
		{
			initialState: {
				embeddedServices: [ 'static-pages' ].map(
					( appDefinitionId ) => {
						return { appDefinitionId };
					}
				),
			},
		},
		false,
		true,
	],
	[
		'personal-website.har',
		{
			initialState: {
				embeddedServices: [ 'static-pages' ].map(
					( appDefinitionId ) => {
						return { appDefinitionId };
					}
				),
			},
		},
		false,
		true,
	],
] )( 'wix: %s', async ( har, config, errorsExpected, logExpected ) => {
	const input = fs.readFileSync( path.join( __dirname, 'fixtures', har ) );

	function stripFirstPubDate( xml ) {
		return xml
			.toString()
			.replace( /<pubDate>.*?<\/pubDate>/, '<pubDate></pubDate>' )
			.trim();
	}

	return getWXRFromWixHAR(
		fetchFromHAR,
		JSON.parse( input ),
		config,
		startExport,
		( url, entry ) => entry
	)
		.then( ( wxrDriver ) => wxrDriver.export() )
		.then( ( xml ) => {
			expect( stripFirstPubDate( xml ) ).toMatchSnapshot();

			if ( logExpected ) {
				expect( console ).toHaveLogged();
			}
			if ( errorsExpected ) {
				expect( console ).toHaveErrored();
			}
		} );
} );
