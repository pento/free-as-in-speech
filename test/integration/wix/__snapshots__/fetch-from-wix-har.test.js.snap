// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wix: empty.har -> empty.wxr 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<pubDate></pubDate>
	<wp:wxr_version>1.2</wp:wxr_version>
</channel>
</rss>"
`;

exports[`wix: wix-basic.har -> wix-basic.wxr 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>

<!-- This is a WordPress eXtended RSS file generated by the \\"Free (as in Speech)\\" browser extension, as an export of your site. -->
<!-- It contains information about your site's posts, pages, comments, categories, and other content. -->
<!-- You may use this file to transfer that content from one site to another. -->
<!-- This file is not intended to serve as a complete backup of your site. -->

<!-- To import this information into a WordPress site follow these steps: -->
<!-- 1. Log in to that site as an administrator. -->
<!-- 2. Go to Tools: Import in the WordPress admin panel. -->
<!-- 3. Install the \\"WordPress\\" importer from the list. -->
<!-- 4. Activate & Run Importer. -->
<!-- 5. Upload this file using the form provided on that page. -->
<!-- 6. You will first be asked to map the authors in this export file to users -->
<!--    on the site. For each author, you may choose to map to an -->
<!--    existing user on the site or to create a new user. -->
<!-- 7. WordPress will then import each of the posts, pages, comments, categories, etc. -->
<!--    contained in this file into your site. -->

<rss version=\\"2.0\\"
	xmlns:excerpt=\\"http://wordpress.org/export/1.2/excerpt/\\"
	xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\"
	xmlns:wfw=\\"http://wellformedweb.org/CommentAPI/\\"
	xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\"
	xmlns:wp=\\"http://wordpress.org/export/1.2/\\"
>
<channel>
	<title><![CDATA[mywebsite]]></title>
	<link><![CDATA[https://12345678.wixsite.com/mywebsite]]></link>
	<pubDate></pubDate>
	<wp:wxr_version>1.2</wp:wxr_version>
	<wp:base_site_url><![CDATA[https://12345678.wixsite.com/mywebsite]]></wp:base_site_url>
	<wp:base_blog_url><![CDATA[https://12345678.wixsite.com/mywebsite]]></wp:base_blog_url>
	<wp:author>
		<wp:author_login><![CDATA[admin]]></wp:author_login>
		<wp:author_display_name><![CDATA[]]></wp:author_display_name>
	</wp:author>
	<item>
		<title><![CDATA[Hello World]]></title>
		<pubDate><![CDATA[Wed, 27 Jan 2021 08:20:41 GMT]]></pubDate>
		<dc:creator><![CDATA[admin]]></dc:creator>
		<guid isPermaLink=\\"false\\"><![CDATA[6676d9d6-0318-4dcc-881f-ba6482cff48d]]></guid>
		<description></description>
		<content:encoded><![CDATA[<!-- wp:paragraph -->
<p>This is my first entry.</p>
<!-- /wp:paragraph -->]]></content:encoded>
		<wp:post_date><![CDATA[2021-01-27 08:20:41]]></wp:post_date>
		<wp:comment_status><![CDATA[open]]></wp:comment_status>
		<wp:status><![CDATA[publish]]></wp:status>
		<wp:post_type><![CDATA[post]]></wp:post_type>
		<wp:is_sticky>0</wp:is_sticky>
	</item>
	<item>
		<title><![CDATA[Draft]]></title>
		<pubDate><![CDATA[]]></pubDate>
		<dc:creator><![CDATA[admin]]></dc:creator>
		<guid isPermaLink=\\"false\\"><![CDATA[1578305e-fcbe-4767-b656-72d409d27bff]]></guid>
		<description></description>
		<content:encoded><![CDATA[]]></content:encoded>
		<wp:post_date><![CDATA[]]></wp:post_date>
		<wp:comment_status><![CDATA[open]]></wp:comment_status>
		<wp:status><![CDATA[draft]]></wp:status>
		<wp:post_type><![CDATA[post]]></wp:post_type>
		<wp:is_sticky>0</wp:is_sticky>
	</item>
</channel>
</rss>"
`;
